html/modules/custom/reliefweb_entities/src/Entity/Report.php

```diff
@@ -249,6 +249,14 @@ public function preSave(EntityStorageInterface $storage) {
 
     // Prepare notifications.
     $this->preparePublicationNotification();
+
+    // Update the entity status based on the user posting rights.
+    $this->updateModerationStatusFromPostingRights();
+
+    // Update the entity status based on the source(s) moderation status.
+    $this->updateModerationStatusFromSourceStatus();
+
+    parent::preSave($storage);
```

Comment:
Needs some tests to confirm that moving the preSave at the end of the function doesn't break anything.

Reply:
See https://github.com/UN-OCHA/rwint9-site/pull/905